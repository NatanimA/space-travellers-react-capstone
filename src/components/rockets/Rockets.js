import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';

import { fetchRockets, toggleReserved } from '../../redux/rockets/rockets';

const Rockets = () => {
  const rockets = useSelector((state) => state.rockets);

  const dispatch = useDispatch();
  useEffect(() => {
    if (rockets.length < 1) {
      dispatch(fetchRockets());
    }
  }, [dispatch, rockets.length]);

  return (
    <section className="section-containers section-rockets">
      <div className="rockets-container">
        {rockets.map((rocket) => (
          <div key={rocket.id} className="rocket__item__container">
            <img src={rocket.image} alt="rocket" />
            <div className="rocket__item__details">
              <h2>
                {rocket.name}
              </h2>
              <p>
                {rocket.reserved ? <span className="reserved__tag">Reserved</span> : null }
                {rocket.description}
              </p>
              <button type="button" id={rocket.id} className={rocket.reserved ? 'cancel__reserved' : 'reserved__btn'} onClick={(e) => dispatch(toggleReserved(e.target.id))}>{rocket.reserved ? 'Cancel Reservation' : 'Reserve Rocket'}</button>
            </div>
          </div>
        ))}
      </div>
    </section>
  );
};

export default Rockets;
