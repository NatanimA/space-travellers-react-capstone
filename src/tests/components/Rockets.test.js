import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Rockets from '../../components/rockets/Rockets';

test('renders rockets component snapshot', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Rockets />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
