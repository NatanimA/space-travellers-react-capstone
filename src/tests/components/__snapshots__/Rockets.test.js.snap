// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rockets component snapshot 1`] = `
<DocumentFragment>
  <section
    class="section-containers section-rockets"
  >
    <div
      class="rockets-container"
    />
  </section>
</DocumentFragment>
`;
